lib/Doctrine/DBAL/Connection.php

```diff
@@ -419,7 +420,7 @@ private function detectDatabasePlatform()
      * or the underlying driver connection cannot determine the platform
      * version without having to query it (performance reasons).
      *
-     * @return string|null
+     * @return string|bool|null
```

Comment:
I'd say it's a bug and should be fixed, not documented.

Reply:
Hmm this is because `SqlAnywhereConnection::getServerVersion()` calls `ResultStatement::fetchColumn()` which returns `string|bool`. :| Replaced with assert there and removed here.